### ISSUE TITLE
Better Temp file management

### ISSUE THREAD

**Issue by stromal** [modin-project/modin#4853]
I wold like to request for better temp file cashing management.
It just fills up my ubuntu 20.04 temp folder.
If storage fills up it should get rid of previously cashed files.

**Comment by vnlitvinov** [comment]
Hello @stromal, thanks for reporting!

Do you have any more data you can provide on those files filling up the `/tmp`? I'm assuming you're talking about Ray files, but would like to confirm that. Would be helpful if you can share at least file names there.

**Comment by stromal** [comment]
yep ray files. I got rid of them and switched off the whole pipeline now. I did not save down the error message but it was like 99% CPU utilization on a 32 thread AWS EC2 instance.

**Comment by stromal** [comment]
@vnlitvinov  

Warning messages similar to this but with 100% CPU, it have crashed at the end and my storage was at 100%
```
distributed.utils_perf - WARNING - full garbage collections took 93% CPU time recently (threshold: 10%)
distributed.utils_perf - WARNING - full garbage collections took 93% CPU time recently (threshold: 10%)
```

**Comment by YarShev** [comment]
@stromal, those warning messages should come from `dask.distributed`. Could you clarify what Modin engine are you using and what Ray files are you seeing?

**Comment by stromal** [comment]
@pyrito I can not put more resources in this. Just make sure that RAY gets rid of the same name variables from it's cash when it generates it's temp files (I got a temp file that was +200 GB, 100% filled up my storage and crashed the whole Ubuntu 20.04 server.)

**Comment by pyrito** [comment]
@stromal I need some more context. Can you show what files are in `tmp/`?

**Comment by stromal** [comment]
@pyrito  that is my issue it takes a long time to reproduce the error. 
It just pushes out this 100% CPU nonsense and the ssd is at 100% in the ubuntu hidden  tmp directory. 
I holder many dataframes with the same name because it went thru for cycles.
If I have time I will let you know here.

**Comment by pyrito** [comment]
@stromal thanks for the context. The `distributed.utils_perf` thing that you are seeing are not from Ray, but Dask. Can you confirm what Modin engine you are using?

**Comment by stromal** [comment]
@pyrito I get back to you tomorrow with that.

**Comment by stromal** [comment]
Versions:

- conda version : 4.6.14
- Python 3.7.12
- modin version: 0.12.1

I have tried installing different modin versions they all crash this is the only package combination that actually functioned:

- https://github.com/modin-project/modin/issues/4906
- https://github.com/modin-project/modin/issues/4719
